rfcs/45-stageless.md

```diff
@@ -496,232 +465,269 @@ They are evaluated a single time during each schedule pass, and the result appli
 
-Yes. This is a large part of the point.
+The word "set" better describes what labeling actually means for scheduling. Sets aren't just an abstract tag, they have a "physical" location in the overall graph (a set is a "supernode" that encapsulates all the nodes inside it). Likewise, sets that intersect (share nodes) can't be made to depend on each other.
 
-The difficulties configuring plugins is a source of [significant user pain](https://github.com/bevyengine/bevy/issues/2160).
-Plugin authors cannot know all of the needs of the end user's app, and so some degree of control must be handed back to the end users.
+### If users can configure system sets, can they mess with systems added by third-party plugins?
 
-This control is carefully limited though: systems can only be configured if they are part of a public set, and all of the systems under a common set are configured as a group.
-In addition, configuration defined by a private set (or directly on systems) cannot be altered, extended or removed by the app: it cannot be named, and so cannot be altered.
+Yes, somewhat, and it's intentional.
 
-This limitation allows plugin authors to carefully design a public API, ensure critical invariants hold, and make serious internal changes without breaking their dependencies (as long as the public sets and their meanings are relatively stable).
+Not being able to schedule plugin systems is a source of [significant user pain](https://github.com/bevyengine/bevy/issues/2160). Instead of the current status quo, we thought a better pattern would be a plugin exporting its top-level set label (at the very least) and the user being responsible for deciding where to put it.
 
-### Why are run criteria shared?
+As it's now possible to define set hierarchies, plugin authors can decide exactly what they want to expose to users in their "public API" *and* still ensure any key logical invariants hold. Users will only be able schedule what the plugin author chooses to make public. Plugin authors could even make massive internal changes without breaking user apps (as long as the "public API" is relatively stable).
 
-Rather than sharing run criteria within a tick, we could evaluate them once for each system.
-This would have a small performance impact, as in practice, almost all run criteria are extremely simple, and fast to check.
+### Why is there no "on-update" system set for states?
 
-However, this behavior is unintuitive, and can result in desynchronization between members with the same run criteria, potentially leading to serious logic errors.
+Conceptually, a set occupies some position in a schedule. However, simply being in a state does not. Any system anywhere can be conditioned to only run if a certain state is active. If there was a builtin "on-update" set, it could only exist in one place. That, along with it being trivial to create a set with the condition that a state must be active, gave little reason to do it.

```

Comment:
```suggestion
Conceptually, a set occupies some position in a schedule.
However, simply being in a state does not.
Any system anywhere can be conditioned to only run if a certain state is active.
If there was a builtin "on-update" set, it could only exist in one place.
That, along with it being trivial to create a set with the condition that a state must be active, gave little reason to do it.
```